### ISSUE TITLE
Errors when running with weback 2

### ISSUE THREAD

**Issue by okonet** [wallabyjs/public#519]
### Issue description or question

I'm seeing following errors when running with webpack 2.1 beta 4

```
wallaby.js started
core v1.0.200
Runtime error: TypeError: setting a property that has only a getter
Runtime error: TypeError: setting a property that has only a getter
at http://localhost:58588/wallaby-webpack.js?1458059246023:1
Runtime error: TypeError: setting a property that has only a getter
Runtime error: TypeError: setting a property that has only a getter
at http://localhost:58588/wallaby-webpack.js?1458059246023:1
```

### Wallaby.js configuration file

```javascript
/* eslint object-shorthand: 0 */
const find = require('lodash/find')
const testRunnerConfig = require('test-runner-config')
const wallabyWebpack = require('wallaby-webpack')
const webpackConfig = require('./webpack/webpack.test')
const testFiles = require('./testFiles')

// removing babel-loader, we will use babel compiler instead, it's more performant
webpackConfig.module.loaders =
    webpackConfig.module.loaders.filter(loader => loader.loader !== 'babel-loader')
delete webpackConfig.devtool

const wallabyPostprocessor = wallabyWebpack(webpackConfig)
const files = testRunnerConfig.getWallabyFiles(testFiles, {
    src: file => ({ pattern: file, instrument: true, load: false, ignore: false }),
    specs: file => ({ pattern: file, instrument: true, load: false, ignore: false }),
    mock: file => ({ pattern: file, instrument: true, load: false, ignore: false })
})
const sourceFiles = find(testFiles, { type: 'src' }).files

module.exports = wallaby => Object.assign(files, {
    postprocessor: wallabyPostprocessor,
    testFramework: 'mocha',
    compilers: sourceFiles.reduce((obj, key) =>
        Object.assign(obj, {
            [key]: wallaby.compilers.babel()
        })
    , {}),
    setup: function () {
        // required to trigger test loading
        window.__moduleBundler.loadTests()
    }
})
```

testFiles.js

```js
module.exports = [
    {
        type: 'lib',
        files: [
            'node_modules/polyfill-function-prototype-bind/bind.js',
            'node_modules/babel-polyfill/dist/polyfill.js',
            'node_modules/sinon/pkg/sinon.js',
            'fixtures/*.*'
        ]
    },
    {
        type: 'mock',
        files: ['fixtures/*.*']
    },
    {
        type: 'src',
        files: [
            'app/**/*.js',
            'server/**/*.js'
        ]
    },
    {
        type: 'specs',
        files: [
            'app/**/*.spec.js',
            'server/**/*.spec.js'
        ]
    }
]
```

.babelrc

```
{
    "presets": [
        "es2015-native-modules",
        "react",
        "stage-1"
    ],
    "plugins": [
        "lodash"
    ]
}
```

package.json

```json
"devDependencies": {
    "chai": "3.5.0",
    "chai-enzyme": "0.4.1",
    "enzyme": "2.1.0",
    "eslint": "^1.10.3",
    "eslint-config-wildbit": "github:wildbit/eslint-config-wildbit",
    "eslint-plugin-react": "^3.16.1",
    "karma": "0.13.22",
    "karma-chrome-launcher": "0.2.2",
    "karma-mocha": "0.2.2",
    "karma-mocha-reporter": "2.0.0",
    "karma-phantomjs-launcher": "1.0.0",
    "karma-sourcemap-loader": "0.3.7",
    "karma-webpack": "1.7.0",
    "lint-staged": "^0.1.1",
    "mocha": "2.4.5",
    "npm-check": "^4.1.4",
    "npm-install-webpack-plugin": "^2.0.0",
    "phantomjs-prebuilt": "2.1.5",
    "polyfill-function-prototype-bind": "0.0.1",
    "pre-commit": "^1.1.2",
    "react-addons-test-utils": "0.14.7",
    "reload-server-webpack-plugin": "1.0.1",
    "stylelint": "3.2.3",
    "stylelint-config-wildbit": "github:wildbit/stylelint-config-wildbit",
    "surge": "0.17.7",
    "test-runner-config": "0.5.0",
    "wallaby-webpack": "0.0.18",
    "webpack-dev-middleware": "1.5.1",
    "webpack-hot-middleware": "2.10.0"
  },
  "dependencies": {
    "apicache": "github:kwhitley/apicache",
    "async-props": "0.2.2",
    "axios": "0.9.1",
    "babel": "6.5.2",
    "babel-cli": "6.6.5",
    "babel-core": "6.7.2",
    "babel-loader": "6.2.4",
    "babel-plugin-lodash": "2.2.1",
    "babel-plugin-react-transform": "2.0.2",
    "babel-plugin-rewire": "0.1.22",
    "babel-plugin-typecheck": "3.8.0",
    "babel-polyfill": "6.7.2",
    "babel-preset-es2015": "6.6.0",
    "babel-preset-es2015-native-modules": "6.6.0",
    "babel-preset-react": "6.5.0",
    "babel-preset-react-hmre": "1.1.1",
    "babel-preset-stage-1": "6.5.0",
    "body-parser": "1.15.0",
    "chalk": "^1.1.1",
    "color": "0.11.1",
    "css-loader": "0.23.1",
    "d3-format": "0.5.1",
    "d3-scale": "0.6.4",
    "d3-time": "0.2.5",
    "d3-time-format": "0.3.1",
    "debug": "2.2.0",
    "dsv-loader": "1.0.3",
    "express": "4.13.4",
    "express-api-helper": "0.0.5",
    "extract-text-webpack-plugin": "1.0.1",
    "file-loader": "0.8.5",
    "html-loader": "0.4.3",
    "html-webpack-plugin": "2.10.0",
    "if-env": "1.0.0",
    "ip": "1.1.2",
    "json-loader": "0.5.4",
    "lodash": "4.6.1",
    "moment": "2.12.0",
    "ngrok": "0.1.99",
    "node-noop": "1.0.0",
    "null-loader": "0.1.1",
    "offline-plugin": "2.0.4",
    "postcss-cssnext": "2.5.1",
    "postcss-focus": "1.0.0",
    "postcss-import": "8.0.2",
    "postcss-loader": "0.8.2",
    "postcss-reporter": "1.3.3",
    "react": "^0.14.6",
    "react-dom": "^0.14.6",
    "react-router": "2.0.1",
    "react-time": "4.0.0",
    "react-tooltip": "1.1.4",
    "reset-css": "2.0.2011012603",
    "source-map-support": "0.4.0",
    "style-loader": "0.13.0",
    "svgo-loader": "1.1.0",
    "url-loader": "0.5.7",
    "victory": "0.4.2",
    "webpack": "2.1.0-beta.4",
    "webpack-merge": "0.8.3"
  }
```

I've tried using webpack babel loader, using `es2015` preset. Nothing helps.

### Code editor or IDE name and version

IntelliJ IDEA v15

### OS name and version

OSX

**Comment by ArtemGovorov** [comment]
Have [created the repo](https://github.com/wallabyjs/wallaby-519) with your files and file structure, but I can't reproduce the issue. It must be happening during one of the project specific files/modules compilation, so may need to ask you to try isolating the source file/module with which the issue can be reproduced. 

Before we go that way, let's try one guess. Open the local `node_modules/wallaby-webpack/index.js` file, replace the line 321 with:

```javascript
var prelude = '(function e(t,n,r){function s(o,u){if(!n[o]){if(!t[o]){var a=typeof require=="function"&&require;if(!u&&a)return a(o,!0);if(i)return i(o,!0);var f=new Error("Cannot find module \'"+o+"\'");throw f.code="MODULE_NOT_FOUND",f}try{var exp={};var l=n[o]={exports:exp,e:exp,id:o,i:o,loaded:false,l:false};}catch(_ex){throw new Error("exp")}var rq=function(e){var n=t[o][1][e];return s(n?n:e)};rq.e=function(a1,a2){a2.call(null,rq)};try{rq.m=tm;rq.c=n;rq.p="";}catch(_ex){throw new Error("rq")}t[o][0].call(exp,rq,l,exp,e,t,n,r);l.l=true}return n[o].exports}var tm={};for(var pr in t){if(t.hasOwnProperty(pr)){tm[pr]=(function(orf){return function(md,mde,wrq){return orf.call(this,wrq,md,mde);}})(t[pr][0]);}}var i=typeof require=="function"&&require;for(var o=0;o<r.length;o++)s(r[o]);return s})';
```
and try running it to see if the error is there and the same.

**Comment by okonet** [comment]
I've tried to change the line but not much have changed. I see

```
wallaby.js started
core v1.0.200
Runtime error: TypeError: '[object Object]' is not a function (evaluating 'keyMirror({
  INSERT_MARKUP: null,
  MOVE_EXISTING: null,
  REMOVE_NODE: null,
  SET_MARKUP: null,
  TEXT_CONTENT: null
})')
Runtime error: TypeError: '[object Object]' is not a function (evaluating 'keyMirror({
  INSERT_MARKUP: null,
  MOVE_EXISTING: null,
  REMOVE_NODE: null,
  SET_MARKUP: null,
  TEXT_CONTENT: null
})')
at http://localhost:60256/__modules/376.js?1458122146843:29
Runtime error: TypeError: '[object Object]' is not a function (evaluating 'fn(this, util)')
at http://localhost:60256/__modules/30.js?1458122146059:40
Runtime error: TypeError: '[object Object]' is not a function (evaluating 'fn(this, util)')
```

First 2 are coming from importing React (AFAIK)

**Comment by okonet** [comment]
It comes from imports/require:

![2016-03-16 at 11 00](https://cloud.githubusercontent.com/assets/11071/13808422/53cf00f8-eb66-11e5-8c79-badb20a3cbe5.png)

It imports my code without any issues:

![2016-03-16 at 11 01](https://cloud.githubusercontent.com/assets/11071/13808448/6e04a414-eb66-11e5-91d7-3a3f87c2c217.png)

**Comment by ArtemGovorov** [comment]
Ok, thanks. Let's try this one (same 321 line):
```
var prelude = '(function e(t,n,r){function s(o,u){if(!n[o]){if(!t[o]){var a=typeof require=="function"&&require;if(!u&&a)return a(o,!0);if(i)return i(o,!0);var f=new Error("Cannot find module \'"+o+"\'");throw f.code="MODULE_NOT_FOUND",f}try{var exp={};var l=n[o]={exports:exp,e:exp,id:o,i:o,loaded:false,l:false};}catch(_ex){throw new Error("exp")}var rq=function(e){var n=t[o][1][e];return s(n?n:e)};rq.e=function(a1,a2){a2.call(null,rq)};try{rq.m=tm;rq.c=n;rq.p="";}catch(_ex){throw new Error("rq")}t[o][0].call(exp,rq,l,exp,e,t,n,r);try{l.exports=l.e=((exp===l.e)?l.exports:l.e);}catch(_ex){throw new Error("ERROR WHILE REASSIGNING EXPORTS")}l.l=true}return n[o].exports}var tm={};for(var pr in t){if(t.hasOwnProperty(pr)){tm[pr]=(function(orf){return function(md,mde,wrq){return orf.call(this,wrq,md,mde);}})(t[pr][0]);}}var i=typeof require=="function"&&require;for(var o=0;o<r.length;o++)s(r[o]);return s})';
```

Does it start printing `ERROR WHILE REASSIGNING EXPORTS`?

**Comment by okonet** [comment]
This last code block seem to fix it.

**Comment by okonet** [comment]
Now I'm getting 

![2016-03-16 at 11 10](https://cloud.githubusercontent.com/assets/11071/13808706/c3ececbe-eb67-11e5-9535-733f2b0fdf43.png)

**Comment by okonet** [comment]
It looks like it has something to do with ES6 export semantics:

```
TypeError: 'undefined' is not an object (evaluating '_d3Interpolate2["default"].values.push')
	at app/components/SystemAvailability/AvailabilityGraph.js:10
	at app/components/SystemAvailability/SystemAvailability.js:10
	at app/components/SystemAvailability/SystemAvailability.spec.js:1

TypeError: 'undefined' is not an object (evaluating '_d3Interpolate2["default"].values.push')
at http://localhost:60806/__modules/808.js?1458123119821:148
```

But these tests are working fine then running with karma + this webpack config:

```
/**
 * TEST WEBPACK CONFIGURATION
 */

const path = require('path')
const webpack = require('webpack')

module.exports = {
    devtool: 'inline-source-map',
    module: {
        // Some libraries don't like being run through babel.
        // If they gripe, put them here.
        noParse: [
            /node_modules\/sinon/,
            /node_modules\/acorn/
        ],
        loaders: [
            { test: /\.json$/, loader: 'json-loader' },
            { test: /\.js$/,
                loader: 'babel-loader',
                exclude: /node_modules/,
                query: {
                    plugins: []
                }
            }
        ]
    },
    plugins: [
        new webpack.NormalModuleReplacementPlugin(/\.(png|svg|scss|css)$/, 'node-noop')
    ],
    // Some node_modules pull in Node-specific dependencies.
    // Since we're running in a browser we have to stub them out. See:
    // https://webpack.github.io/docs/configuration.html#node
    // https://github.com/webpack/node-libs-browser/tree/master/mock
    // https://github.com/webpack/jade-loader/issues/8#issuecomment-55568520
    node: {
        fs: 'empty',
        child_process: 'empty',
        net: 'empty',
        tls: 'empty'
    },
    // required for enzyme to work properly
    externals: {
        jsdom: 'window',
        cheerio: 'window',
        sinon: 'sinon',
        'react/lib/ExecutionEnvironment': true,
        'react/lib/ReactContext': 'window'
    },
    resolve: {
        modules: [
            path.resolve(__dirname, '..', 'fixtures'),
            'node_modules'
        ],
        alias: {
            // required for enzyme to work properly
            sinon: 'sinon/pkg/sinon'
        }
    }
}
```

**Comment by ArtemGovorov** [comment]
Ok, let's do one step at a time. 

I still don't understand the reason for the original error and the second one may just be a result of my experimenting with the loader line.

Try this one:
```javascript
var prelude = '(function e(t,n,r){function s(o,u){if(!n[o]){if(!t[o]){var a=typeof require=="function"&&require;if(!u&&a)return a(o,!0);if(i)return i(o,!0);var f=new Error("Cannot find module \'"+o+"\'");throw f.code="MODULE_NOT_FOUND",f}try{var exp={};var l=n[o]={exports:exp,e:exp,id:o,i:o,loaded:false,l:false};}catch(_ex){throw new Error("exp")}var rq=function(e){var n=t[o][1][e];return s(n?n:e)};rq.e=function(a1,a2){a2.call(null,rq)};try{rq.m=tm;rq.c=n;rq.p="";}catch(_ex){throw new Error("rq")}t[o][0].call(exp,rq,l,exp,e,t,n,r);l.exports=l.e=((exp===l.e)?l.exports:l.e);try{l.loaded=true;l.l=true}catch(_ex){throw new Error("ERROR WHILE REASSIGNING loaded")}}return n[o].exports}var tm={};for(var pr in t){if(t.hasOwnProperty(pr)){tm[pr]=(function(orf){return function(md,mde,wrq){return orf.call(this,wrq,md,mde);}})(t[pr][0]);}}var i=typeof require=="function"&&require;for(var o=0;o<r.length;o++)s(r[o]);return s})';
```

I'll try to import some of the modules from your package.json in this sample - https://github.com/wallabyjs/wallaby-519 to see if I can reproduce the issue.

**Comment by okonet** [comment]
```
wallaby.js started
core v1.0.200
Runtime error: Error: ERROR WHILE REASSIGNING loaded
Runtime error: Error: ERROR WHILE REASSIGNING loaded
at http://localhost:61062/wallaby-webpack.js?1458123746897:1
console.log: [Function]
console.log: [Function]
console.log: [Function]
console.log: [Function]
console.log: [Function]
Runtime error: Error: ERROR WHILE REASSIGNING loaded
console.log: [Function]
console.log: [Function]
console.log: [Function]
console.log: [Function]
console.log: [Function]
Runtime error: Error: ERROR WHILE REASSIGNING loaded
at http://localhost:61062/wallaby-webpack.js?1458123746897:1
```

**Comment by ArtemGovorov** [comment]
Ok thanks, that clarifies the issue, but I'm not yet sure why it happens and the second issue doesn't seem to be related. Let me try to add your webpack config and a test like yours to the https://github.com/wallabyjs/wallaby-519 sample to see if I can reproduce it.

**Comment by ArtemGovorov** [comment]
So I have added your webpack config as is to the https://github.com/wallabyjs/wallaby-519  sample and it still doesn't throw any errors. Could you please try to see if it works for you?

Since the second error is thrown from the `AvailabilityGraph.js` using d3 module, could you please share an example of how you're importing/`require`-ing the module from the file, so I could try it in the sample?

**Comment by ArtemGovorov** [comment]
Ok, I managed to reproduce the **original issue**: `Runtime error: TypeError: setting a property that has only a getter`  in another project and **fixed it in the latest `wallaby-webpack@0.0.19`**.

But I can't reproduce the **second one** (the one that has something to do with ES6 export semantics).

The `d3Interpolate` in the error means that you are importing something from from the `d3-scale` in the `AvailabilityGraph.js`. I have tried importing in a [couple of ways](https://github.com/wallabyjs/wallaby-519/blob/master/app/a.spec.js#L4), but no error.

Then the error itself:
```
TypeError: 'undefined' is not an object (evaluating '_d3Interpolate2["default"].values.push')
```
may mean that you are running with `es2015` preset, not the `es2015-native-modules` like you have shared in the original `.babelrc`. Is it correct? It would be better if my environment was exactly like your environment, clearly it's not because I can't reproduce it yet. Anyway, I have tried both `es2015` and `es2015-native-modules` and can't reproduce the error with any of the presets.

It may help if you could share a sample how you import `d3-scale` in the `AvailabilityGraph.js`.

**Comment by joergwasmeier** [comment]
```
TypeError: Object is not a constructor (evaluating 'keyMirror({
  INSERT_MARKUP: null,
  MOVE_EXISTING: null,
  REMOVE_NODE: null,
  SET_MARKUP: null,
  TEXT_CONTENT: null
})')
at http://localhost:52881/__modules/79.js?1458495934521:24
```

I have also the problem with webpack 1 it works fine but when i switch it to beta 2.1.0-beta.4 i have this error. It cames from react i thing.

**Comment by ArtemGovorov** [comment]
@joergwasmeier What version of `wallaby-webpack` package do you have installed? The error used to happen in the older versions of the package, try upgrading to the latest.

**Comment by okonet** [comment]
I've update to the latest `wallaby-webpack` 0.0.21 and the original error is gone as well as d3 related one.

But now I have this error:

```
server/api/delivery/cleanupDeliveryData.spec.js cleanupDeliveryData should return array [0 ms]

	instanceof called on an object with an invalid prototype property.
		at node_modules/polyfill-function-prototype-bind/bind.js:14
		at server/api/delivery/cleanupDeliveryData.js:7
		at node_modules/polyfill-function-prototype-bind/bind.js:17
		at server/api/delivery/cleanupDeliveryData.spec.js:39
```

**Comment by okonet** [comment]
Nevermind my last message. It seems the issue in my code.

**Comment by ArtemGovorov** [comment]
Thanks for providing the error stack!
You may try another polyfill: https://www.npmjs.com/package/phantomjs-polyfill, I remember some users having issues with other polyfills (not sure about the one you're using).

**Comment by ArtemGovorov** [comment]
@okonet Ah, ok, sorry have sent the message before saw yours. Anyway, if you still have issues with the polyfill, maybe trying the [`phantomjs-polyfill`](https://www.npmjs.com/package/phantomjs-polyfill) may help.

**Comment by okonet** [comment]
@ArtemGovorov the other polyfill works. Thanks!

**Comment by okonet** [comment]
It's probably worth putting into FAQ or home page.

**Comment by ArtemGovorov** [comment]
Great.
We [do recommend the polyfill on the FAQ](http://wallabyjs.com/docs/intro/troubleshooting.html#undefined-is-not-a-function-evaluating-bind). I'll add a note about other polyfills that may not work in PhantomJs.
Can the issue be closed?

**Comment by okonet** [comment]
Yes, it looks like this. Thanks again!

**Comment by ArtemGovorov** [comment]
Thanks. 

Regarding the polyfill, I just though of something and I have no idea why I didn't think about it before. So I just included the correct polyfill into some script that wallaby loads anyway. This way you don't need to bother installing the package and add the reference to the files list - it should just work. Available in the core v1.0.207 (have just published it).
